src/python/pose_format/bin/directory.py

```diff
@@ -1,45 +1,171 @@
+    )
+    parser.add_argument(
+        "-r",
+        "--recursive",
+        action="store_true",
+        help="Whether to search for videos recursively",
+    )
+    parser.add_argument(
+        "--keep-video-suffixes",
+        action="store_true",
+        help="Whether to drop the video extension (output for foo.mp4 becomes foo.pose, and foo.webm ALSO becomes foo.pose) or append to it (foo.mp4 becomes foo.mp4.pose, foo.webm output is foo.webm.pose). If there are multiple videos with the same basename but different extensions, this will create a .pose file for each. Otherwise only the first video will be posed.",
+    )
+    parser.add_argument(
+        "--video-suffixes",
+        type=str,
+        choices=_SUPPORTED_VIDEO_FORMATS,
+        default=_SUPPORTED_VIDEO_FORMATS,
+        help="Video extensions to search for. Defaults to searching for all supported.",
+    )
+    parser.add_argument(
+        "--additional-config",
+        type=str,
+        help="additional configuration for the pose estimator",
+    )
     args = parser.parse_args()
 
-    missing_pose_files = find_missing_pose_files(args.directory)
+    videos_with_missing_pose_files = find_videos_with_missing_pose_files(
+        args.directory,
+        video_suffixes=args.video_suffixes,
+        recursive=args.recursive,
+        keep_video_suffixes=args.keep_video_suffixes,
+    )
+
+    print(f"Found {len(videos_with_missing_pose_files)} videos missing pose files.")
+
+    pose_files_that_will_be_created = set(
+        [
+            get_corresponding_pose_path(vid_path, args.keep_video_suffixes)
+            for vid_path in videos_with_missing_pose_files
+        ]
+    )
+    if len(pose_files_that_will_be_created) < len(videos_with_missing_pose_files):
+        continue_input = input(
+            f"With current naming strategy (without --keep-video-suffixes), name collisions will result in only {len(pose_files_that_will_be_created)} .pose files being created. Continue? [y/n]"
+        )
+        if continue_input.lower() != "y":
+            print(f"Exiting. To keep video suffixes and , use --keep-video-suffixes")
+            exit()
+
     additional_config = parse_additional_config(args.additional_config)
 
-    for mp4_path in tqdm(missing_pose_files):
-        pose_file_name = removesuffix(mp4_path, ".mp4") + ".pose"
-        pose_video(mp4_path, pose_file_name, args.format, additional_config)
+    pose_with_no_errors = []
+
+    for vid_path in tqdm(videos_with_missing_pose_files):
+        try:
+            pose_path = get_corresponding_pose_path(
+                video_path=vid_path, keep_video_suffixes=args.keep_video_suffixes
+            )
+            if pose_path.is_file():
+                print(f"Skipping {vid_path}, corresponding .pose file already created.")
+                continue
+            pose_video(vid_path, pose_path, args.format, additional_config)
+            pose_with_no_errors.append(vid_path)
```

Comment:
if this is basically a counter, just count - `+=1` no need to create another array

Reply:
I had thought to use it at a later date for more detailed logging/debugging, e.g. write out the specific ones that didn't work. But I can just add the list then when/if it's needed.